pkg/codegen/nodejs/gen.go

```diff
@@ -1000,7 +1009,15 @@ func (mod *modContext) getTypeImportsForResource(t schema.Type, recurse bool, ex
 		// If it's from another package, add an import for the external package.
 		if t.Package != nil && t.Package != mod.pkg {
 			pkg := t.Package.Name
-			externalImports.Add(fmt.Sprintf("import * as %[1]s from \"@pulumi/%[1]s\";", pkg))
+			var nodePackageInfo NodePackageInfo
+			if languageInfo, hasLanguageInfo := mod.pkg.Language["nodejs"]; hasLanguageInfo {
+				nodePackageInfo = languageInfo.(NodePackageInfo)
+			}
+			if imp, ok := nodePackageInfo.ProviderNameToModuleName[pkg]; ok {
+				externalImports.Add(fmt.Sprintf("import * as %[1]s from \"%[2]s\";", fmt.Sprintf("pulumi%s", title(pkg)), imp))
+			} else {
+				externalImports.Add(fmt.Sprintf("import * as %[1]s from \"@pulumi/%[2]s\";", fmt.Sprintf("pulumi%s", title(pkg)), pkg))
+			}
```

Comment:
Nit: The index args aren't necessary, right?

```suggestion
			if imp, ok := nodePackageInfo.ProviderNameToModuleName[pkg]; ok {
				externalImports.Add(fmt.Sprintf("import * as %s from \"%s\";", fmt.Sprintf("pulumi%s", title(pkg)), imp))
			} else {
				externalImports.Add(fmt.Sprintf("import * as %s from \"@pulumi/%s\";", fmt.Sprintf("pulumi%s", title(pkg)), pkg))
			}
```

Reply:
They're not strictly necessary - they were carried over from somewhere else. They can be removed.

Reply:
Addressed in 88c9fdf.